### PR TITLE
Add zli close unit tests + system test

### PR DESCRIPTION
**Pull request by sidpremkumar**
## Description of the change

Adds test for our close handler functions

## Testing

Describe how to test this PR....

**backend branch:** 
**bzero branch:** 

#### Ready to run system tests?

- [x] Yes

## Relevant release note information

Release Notes: Adds unit test on zli close along with system tests

## Related JIRA tickets

Relates to JIRA: CWC-1687

## Relevant HotFix if applicable

Relevant HotFix PR Number:

## Have you considered the security impacts?

Does this PR have any security impact?

- [ ] Yes
- [ ] No

If yes, please explain:

### CODE REVIEW

**Review by chad-cwc — COMMENTED** [review]
I'm struggling a bit with these tests since all they do is expect functions to have been or not to have been called. I'm not saying there's no value in that - there definitely can be.

I guess it comes down to this: Can these tests be helpful in preventing a regression from being introduced into the code? I'm not sure. Maybe?

If it would be helpful, I could take a second pass through and mark the expects that I'm most doubtful of - like the listSpaces one. Or we could just talk through it.

File: src/handlers/close-connection/close-connection.handler.test.ts [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+        // Assert we didnt call any getConnection call
+        expect(getConnectionSpy).not.toHaveBeenCalled();
+
+        // Ensure we called list spaces
+        expect(listSpacesSpy).toHaveBeenCalled();
+
+        // Expect that we have not called the following
+        expect(closeSpaceSpy).toHaveBeenCalledWith(mockSpaceSummary.id);
+        expect(createSpaceSpy).toHaveBeenCalledWith('cli-space');
+        expect(closeConnectionSpy).not.toHaveBeenCalled();
```
Comment (chad-cwc) [inline src/handlers/close-connection/close-connection.handler.test.ts:54]: I think we should look at these expects as what is essential for `zli close -a`, and to that end, I think we could remove at least:

`expect(listSpacesSpy).toHaveBeenCalled();`
`expect(createSpaceSpy).toHaveBeenCalledWith('cli-space');`

But, if you disagree, it's not worth holding up getting this in.

Comment (sidpremkumar) [reply src/handlers/close-connection/close-connection.handler.test.ts]: https://github.com/bastionzero/zli/pull/387/commits/bc172df8cf3495bf3370753ab0b32c1cffde274c

File: src/handlers/close-connection/close-connection.handler.test.ts [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+        });
+        expect(cleanExitSpy).toHaveBeenCalledWith(0, expect.anything());
+
+        // Assert we didnt call any getConnection call
+        expect(getConnectionSpy).not.toHaveBeenCalled();
+
+        // Ensure we called list spaces
+        expect(listSpacesSpy).toHaveBeenCalled();
+
+        // Expect that we have not called the following
```
[the remark this answers is not in the document]
Comment (sidpremkumar) [reply src/handlers/close-connection/close-connection.handler.test.ts]: I forgot which commit this was fixed, but has since been fixed 🔥

**Review by chad-cwc — APPROVED** [review]

**Pull request merged**